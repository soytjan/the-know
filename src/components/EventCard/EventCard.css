@import url("https://fonts.googleapis.com/css?family=Open+Sans:300,400|Poiret+One");
.EventCard {
  max-width: 280px;
  min-height: 450px;
  background-color: #FBFCFC;
  line-height: 140%;
  display: flex;
  flex-direction: column; }
  .EventCard a {
    text-decoration: none;
    border: 1px solid #ECF0F1;
    padding: 5px 8px;
    color: #EC5078;
    border-radius: 10px; }
    .EventCard a:hover {
      background-color: #BEECE9; }

.btn-cont {
  padding: 5px;
  height: 50px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 170px;
  margin: 0 auto; }

.btn-fav {
  font-size: 1.3em;
  border: 1px solid #ECF0F1;
  color: #EC5078;
  border-radius: 10px; }
  .btn-fav:hover {
    background-color: #BEECE9; }

.favorited button {
  background-color: #F8BFCD; }

.img-event {
  height: 128px;
  width: 128px; }

.img-box {
  background-color: #BEECE9;
  height: 280px;
  width: 280px;
  display: flex;
  justify-content: center;
  align-items: center;
  border-bottom: 5px solid #4ECDC4; }

.event-info {
  padding: 10px;
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: space-between; }

.event {
  background: center/contain no-repeat url("../../assets/event.jpg"); }

.food {
  background: center/contain no-repeat url("../../assets/food.jpg"); }

.music {
  background: center/contain no-repeat url("../../assets/music.jpg"); }

.culture {
  background: center/contain no-repeat url("../../assets/culture.jpg"); }

.nightlife {
  background: center/contain no-repeat url("../../assets/nightlife.jpg"); }

.search {
  background: center/contain no-repeat url("../../assets/event.jpg"); }
