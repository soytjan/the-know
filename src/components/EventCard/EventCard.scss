@import "../../variables.scss";

.EventCard {
  border: 1px solid turquoise;
}

.favorited {
  background-color: magenta;
}

.img-event {
  height: 128px;
  width: 128px;
}

.img-box {
  background-color: $light-turq;
  height: 230px;
  width: 230px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.event {
  background: center / contain no-repeat url('../../assets/event.jpg');
}

.food {
  background: center / contain no-repeat url('../../assets/food.jpg');
}

.music {
  background: center / contain no-repeat url('../../assets/music.jpg');
}

.culture {
  background: center / contain no-repeat url('../../assets/culture.jpg');
}

.nightlife {
  background: center / contain no-repeat url('../../assets/nightlife.jpg');
}

.search {
  background: center / contain no-repeat url('../../assets/event.jpg')
}