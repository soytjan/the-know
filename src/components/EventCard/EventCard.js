import React from 'react';
import PropTypes from 'prop-types';
// import event from '../../assets/event.jpg';
// import food from '../../assets/food.jpg';
// import music from '../../assets/music.jpg';
// import culture from '../../assets/culture.jpg';
// import nightlife from '../../assets/nightlife.jpg';
import './EventCard.css';

const EventCard = (props) => {
  const { event, type, onFavorite } = props;
  const favorited = event.isFavorited ? 'favorited' : '';
  const img = event.image ? 
    <img className='img-event' src={event.image.medium.url} alt={event.title} /> 
    : <div className={`img-event ${type}`}></div>;

  return (
    <article className={`EventCard ${favorited}`}>
      <div className={`img-box`}>
        {img}
      </div>
      <div>
        <h4>{event.title}</h4>
        <p>{event.venueName}</p>
        <p>{event.startTime}</p>
        <div className='btn-cont'>
          <button 
            className='btn-fav'
            onClick={() => onFavorite(event, type)}
          >
            <i className="far fa-heart"></i>
          </button>
          <button className='btn-see-more'>SEE MORE</button>
        </div>
      </div>
    </article>
  )
}

EventCard.propTypes = {
  event: PropTypes.object,
  type: PropTypes.string,
  onFavorite: PropTypes.func
};

export default EventCard;
