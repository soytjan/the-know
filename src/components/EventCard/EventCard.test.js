import React from 'react';
import { shallow } from 'enzyme';
import EventCard from './EventCard';

describe('EventCard', () => {
  let renderedComponent;
  let mockEvent;
  let mockOnFavorite;

  beforeAll(() => {
    mockEvent = {
      title: "Thursday Work Day, July 5 - Multiple Neighborhoods",
      description: "event description",
      category: "music",
      image: null,
      url: "fake.com",
      id: "E0-001-111209195-8"
    };
  });

  beforeEach(() => {
    mockOnFavorite = jest.fn();
    renderedComponent = shallow(
      <EventCard
        event={mockEvent}
        onFavorite={mockOnFavorite}
        key={1}
      />
    );
  });

  it('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should reflect UI change when event is favorited', () => {
    const mockEvent = {
      title: "Thursday Work Day, July 5 - Multiple Neighborhoods",
      description: "event description",
      category: "music",
      image: null,
      url: "fake.com",
      id: "E0-001-111209195-8",
      isFavorited: true
    };
    const renderedComponent = shallow(
      <EventCard
        event={mockEvent}
        onFavorite={mockOnFavorite}
        key={1}
      />
    );

    expect(renderedComponent).toMatchSnapshot();
  });

  it('should call onFavorite from props when button is clicked', () => {
    renderedComponent.find('button').simulate('click');

    expect(mockOnFavorite).toHaveBeenCalled();
  });
});