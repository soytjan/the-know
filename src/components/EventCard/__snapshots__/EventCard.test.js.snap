// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventCard
    event={
        Object {
            "category": "music",
            "description": "event description",
            "id": "E0-001-111209195-8",
            "image": null,
            "title": "Thursday Work Day, July 5 - Multiple Neighborhoods",
            "url": "fake.com",
          }
    }
    onFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="img-box"
>
          <div
                    className="img-event music2"
          />
</div>,
        <div
          className="event-info"
>
          <div
                    className="event-info-text"
          >
                    <h4>
                              Thursday Work Day, July 5 - Multiple Neighborhoods
                    </h4>
                    <p />
                    <p />
          </div>
          <div
                    className="btn-cont"
          >
                    <button
                              className="btn-fav"
                              onClick={[Function]}
                    >
                              <i
                                        className="far fa-heart"
                              />
                    </button>
                    <a
                              href="fake.com"
                              target="_blank"
                    >
                              SEE MORE
                    </a>
          </div>
</div>,
      ],
      "className": "EventCard ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="img-event music2"
/>,
          "className": "img-box",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "img-event music2",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="event-info-text"
>
              <h4>
                            Thursday Work Day, July 5 - Multiple Neighborhoods
              </h4>
              <p />
              <p />
</div>,
            <div
              className="btn-cont"
>
              <button
                            className="btn-fav"
                            onClick={[Function]}
              >
                            <i
                                          className="far fa-heart"
                            />
              </button>
              <a
                            href="fake.com"
                            target="_blank"
              >
                            SEE MORE
              </a>
</div>,
          ],
          "className": "event-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Thursday Work Day, July 5 - Multiple Neighborhoods
</h4>,
                <p />,
                <p />,
              ],
              "className": "event-info-text",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Thursday Work Day, July 5 - Multiple Neighborhoods",
                },
                "ref": null,
                "rendered": "Thursday Work Day, July 5 - Multiple Neighborhoods",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="btn-fav"
                  onClick={[Function]}
>
                  <i
                                    className="far fa-heart"
                  />
</button>,
                <a
                  href="fake.com"
                  target="_blank"
>
                  SEE MORE
</a>,
              ],
              "className": "btn-cont",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="far fa-heart"
/>,
                  "className": "btn-fav",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "far fa-heart",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "SEE MORE",
                  "href": "fake.com",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "SEE MORE",
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="img-box"
>
            <div
                        className="img-event music2"
            />
</div>,
          <div
            className="event-info"
>
            <div
                        className="event-info-text"
            >
                        <h4>
                                    Thursday Work Day, July 5 - Multiple Neighborhoods
                        </h4>
                        <p />
                        <p />
            </div>
            <div
                        className="btn-cont"
            >
                        <button
                                    className="btn-fav"
                                    onClick={[Function]}
                        >
                                    <i
                                                className="far fa-heart"
                                    />
                        </button>
                        <a
                                    href="fake.com"
                                    target="_blank"
                        >
                                    SEE MORE
                        </a>
            </div>
</div>,
        ],
        "className": "EventCard ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="img-event music2"
/>,
            "className": "img-box",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "img-event music2",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="event-info-text"
>
                <h4>
                                Thursday Work Day, July 5 - Multiple Neighborhoods
                </h4>
                <p />
                <p />
</div>,
              <div
                className="btn-cont"
>
                <button
                                className="btn-fav"
                                onClick={[Function]}
                >
                                <i
                                                className="far fa-heart"
                                />
                </button>
                <a
                                href="fake.com"
                                target="_blank"
                >
                                SEE MORE
                </a>
</div>,
            ],
            "className": "event-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    Thursday Work Day, July 5 - Multiple Neighborhoods
</h4>,
                  <p />,
                  <p />,
                ],
                "className": "event-info-text",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Thursday Work Day, July 5 - Multiple Neighborhoods",
                  },
                  "ref": null,
                  "rendered": "Thursday Work Day, July 5 - Multiple Neighborhoods",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="btn-fav"
                    onClick={[Function]}
>
                    <i
                                        className="far fa-heart"
                    />
</button>,
                  <a
                    href="fake.com"
                    target="_blank"
>
                    SEE MORE
</a>,
                ],
                "className": "btn-cont",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      className="far fa-heart"
/>,
                    "className": "btn-fav",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "far fa-heart",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "SEE MORE",
                    "href": "fake.com",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "SEE MORE",
                  "type": "a",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`EventCard should reflect UI change when event is favorited 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventCard
    event={
        Object {
            "category": "music",
            "description": "event description",
            "id": "E0-001-111209195-8",
            "image": null,
            "isFavorited": true,
            "title": "Thursday Work Day, July 5 - Multiple Neighborhoods",
            "url": "fake.com",
          }
    }
    onFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="img-box"
>
          <div
                    className="img-event music3"
          />
</div>,
        <div
          className="event-info"
>
          <div
                    className="event-info-text"
          >
                    <h4>
                              Thursday Work Day, July 5 - Multiple Neighborhoods
                    </h4>
                    <p />
                    <p />
          </div>
          <div
                    className="btn-cont"
          >
                    <button
                              className="btn-fav"
                              onClick={[Function]}
                    >
                              <i
                                        className="far fa-heart"
                              />
                    </button>
                    <a
                              href="fake.com"
                              target="_blank"
                    >
                              SEE MORE
                    </a>
          </div>
</div>,
      ],
      "className": "EventCard favorited",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="img-event music3"
/>,
          "className": "img-box",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "img-event music3",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="event-info-text"
>
              <h4>
                            Thursday Work Day, July 5 - Multiple Neighborhoods
              </h4>
              <p />
              <p />
</div>,
            <div
              className="btn-cont"
>
              <button
                            className="btn-fav"
                            onClick={[Function]}
              >
                            <i
                                          className="far fa-heart"
                            />
              </button>
              <a
                            href="fake.com"
                            target="_blank"
              >
                            SEE MORE
              </a>
</div>,
          ],
          "className": "event-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Thursday Work Day, July 5 - Multiple Neighborhoods
</h4>,
                <p />,
                <p />,
              ],
              "className": "event-info-text",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Thursday Work Day, July 5 - Multiple Neighborhoods",
                },
                "ref": null,
                "rendered": "Thursday Work Day, July 5 - Multiple Neighborhoods",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="btn-fav"
                  onClick={[Function]}
>
                  <i
                                    className="far fa-heart"
                  />
</button>,
                <a
                  href="fake.com"
                  target="_blank"
>
                  SEE MORE
</a>,
              ],
              "className": "btn-cont",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="far fa-heart"
/>,
                  "className": "btn-fav",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "far fa-heart",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "SEE MORE",
                  "href": "fake.com",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "SEE MORE",
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="img-box"
>
            <div
                        className="img-event music3"
            />
</div>,
          <div
            className="event-info"
>
            <div
                        className="event-info-text"
            >
                        <h4>
                                    Thursday Work Day, July 5 - Multiple Neighborhoods
                        </h4>
                        <p />
                        <p />
            </div>
            <div
                        className="btn-cont"
            >
                        <button
                                    className="btn-fav"
                                    onClick={[Function]}
                        >
                                    <i
                                                className="far fa-heart"
                                    />
                        </button>
                        <a
                                    href="fake.com"
                                    target="_blank"
                        >
                                    SEE MORE
                        </a>
            </div>
</div>,
        ],
        "className": "EventCard favorited",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="img-event music3"
/>,
            "className": "img-box",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "img-event music3",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="event-info-text"
>
                <h4>
                                Thursday Work Day, July 5 - Multiple Neighborhoods
                </h4>
                <p />
                <p />
</div>,
              <div
                className="btn-cont"
>
                <button
                                className="btn-fav"
                                onClick={[Function]}
                >
                                <i
                                                className="far fa-heart"
                                />
                </button>
                <a
                                href="fake.com"
                                target="_blank"
                >
                                SEE MORE
                </a>
</div>,
            ],
            "className": "event-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    Thursday Work Day, July 5 - Multiple Neighborhoods
</h4>,
                  <p />,
                  <p />,
                ],
                "className": "event-info-text",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Thursday Work Day, July 5 - Multiple Neighborhoods",
                  },
                  "ref": null,
                  "rendered": "Thursday Work Day, July 5 - Multiple Neighborhoods",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="btn-fav"
                    onClick={[Function]}
>
                    <i
                                        className="far fa-heart"
                    />
</button>,
                  <a
                    href="fake.com"
                    target="_blank"
>
                    SEE MORE
</a>,
                ],
                "className": "btn-cont",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      className="far fa-heart"
/>,
                    "className": "btn-fav",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "far fa-heart",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "SEE MORE",
                    "href": "fake.com",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "SEE MORE",
                  "type": "a",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
